docs/docs/routing.md

```diff
@@ -0,0 +1,238 @@
+---
+id: routing
+title: Routing
+permalink: docs/routing.html
+---
+
+Routing is the process of choosing what to render based on the browser's current URL.
+
+The simplest way to implement Routing in a React application is with a top-level Component that chooses which content to render by URL. For example, a top-level `App` component may decide what to render based on [window.location.pathname](https://developer.mozilla.org/en-US/docs/Web/API/Location):
+
+```js
+function Menu() {
+  return (
+    <div>
+      <a href="/news">News</a>
+      <a href="/profile">Profile</a>
+    </div>
+  );
+}
+
+function App() {
+  const location = window.location;
+
+  let content;
+  if (location.pathname == '/news') {
+    content = <h1>News</h1>;
+  } else if (location.pathname == '/profile') {
+    content = <h1>Profile</h1>;
+  } else {
+    content = <h1>Not Found</h1>;
+  }
+  
+  return (
+    <div>
+      <Menu />
+      <div>{content}</div>
+    </div>
+  );
+}
+```
+
+But while this app will work, it has one major issue; it will reload the entire page every time a user clicks a link.
+
+## HTML5 History
+
+The HTML5 [History API](https://developer.mozilla.org/en/docs/Web/API/History) makes it possible to update the browser's URL without causing a page reload. It also emits browser events when the user clicks *Back* or *Forward*. By using this API, you can write applications that respond faster and behave as expected when *Back* or *Forward* are clicked.
+
+Single page applications will often make use of the following two parts of the History API:
+
+- The [popstate](https://developer.mozilla.org/en-US/docs/Web/Events/popstate) browser event, which is emitted when *Back* or *Forward* are clicked
```

Comment:
Maybe say "the `popstate` browser event on the `window` object" to make it clear how to subscribe to it.